src/book/book.rs

```diff
@@ -77,10 +77,10 @@ fn create_missing(src_dir: &Path, summary: &Summary) -> Result<()> {
 /// [`iter()`]: #method.iter
 /// [`for_each_mut()`]: #method.for_each_mut
 #[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
 pub struct Book {
     /// The sections in this book.
     pub sections: Vec<BookItem>,
-    __non_exhaustive: (),
```

Comment:
This is a breaking change, can you revert it please?

Reply:
Isn't the added `#[non_exhaustive]` attribute the same as the private field?

Reply:
No, it is exposed to preprocessors. There was a similar attempt earlier which was reverted: https://github.com/rust-lang/mdBook/pull/1848

Reply:
I removed the change.